File: android/app/src/main/kotlin/com/ledgerflow/models/TransactionModels.kt
package com.ledgerflow.models

import java.io.Serializable
import java.util.*

data class Transaction(
    val id: String = UUID.randomUUID().toString(),
    val userId: String,
    val type: String, // "income" or "expense"
    val category: String,
    val amount: Double,
    val currency: String,
    val description: String,
    val date: Date = Date(),
    val attachmentUrl: String? = null,
    val createdAt: Date = Date()
) : Serializable {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "id" to id,
            "userId" to userId,
            "type" to type,
            "category" to category,
            "amount" to amount,
            "currency" to currency,
            "description" to description,
            "date" to date,
            "attachmentUrl" to attachmentUrl,
            "createdAt" to createdAt
        )
    }
}

data class MonthlyReport(
    val month: String,
    val totalIncome: Double,
    val totalExpense: Double,
    val balance: Double,
    val categories: List<String>
) : Serializable

File: ios/LedgerFlow/Models/TransactionModels.swift
import Foundation

struct Transaction: Codable {
    let id: String
    let userId: String
    let type: String // "income" or "expense"
    let category: String
    let amount: Double
    let currency: String
    let description: String
    let date: Date
    let attachmentUrl: String?
    let createdAt: Date
    
    init(
        id: String = UUID().uuidString,
        userId: String,
        type: String,
        category: String,
        amount: Double,
        currency: String,
        description: String,
        date: Date = Date(),
        attachmentUrl: String? = nil,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.userId = userId
        self.type = type
        self.category = category
        self.amount = amount
        self.currency = currency
        self.description = description
        self.date = date
        self.attachmentUrl = attachmentUrl
        self.createdAt = createdAt
    }
    
    func toDictionary() -> [String: Any] {
        var dict: [String: Any] = [
            "id": id,
            "userId": userId,
            "type": type,
            "category": category,
            "amount": amount,
            "currency": currency,
            "description": description,
            "date": date,
            "createdAt": createdAt
        ]
        if let url = attachmentUrl {
            dict["attachmentUrl"] = url
        }
        return dict
    }
}

struct MonthlyReport: Codable {
    let month: String
    let totalIncome: Double
    let totalExpense: Double
    let balance: Double
    let categories: [String]
}
